File: tests/fixtures/objc/simple_class.m
//
//  MyClass.m
//  Example
//
//  Simple Objective-C class implementation
//

#import "simple_class.h"

@implementation MyClass

- (instancetype)initWithName:(NSString *)name age:(NSInteger)age {
    self = [super init];
    if (self) {
        _name = name;
        _age = age;
    }
    return self;
}

- (NSString *)greet {
    return [NSString stringWithFormat:@"Hello, I'm %@ and I'm %ld years old", 
            self.name, (long)self.age];
}

+ (instancetype)personWithName:(NSString *)name {
    return [[MyClass alloc] initWithName:name age:0];
}

- (void)privateMethod {
    NSLog(@"This is a private method");
}

@end

File: tests/fixtures/swift/simple_class.swift
import Foundation

/// A simple user class demonstrating Swift class features
class User {
    /// The user's name
    var name: String
    
    /// The user's age
    var age: Int
    
    /// Computed property for display name
    var displayName: String {
        return "User: \(name)"
    }
    
    /// Property with observers
    var status: String = "active" {
        willSet {
            print("Status will change to \(newValue)")
        }
        didSet {
            print("Status changed from \(oldValue)")
        }
    }
    
    /// Initializer
    init(name: String, age: Int) {
        self.name = name
        self.age = age
    }
    
    /// Greet method
    func greet() -> String {
        return "Hello, I'm \(name)"
    }
    
    /// Update age method
    func updateAge(_ newAge: Int) {
        self.age = newAge
    }
}

/// Subclass demonstrating inheritance
class AdminUser: User {
    var permissions: [String]
    
    init(name: String, age: Int, permissions: [String]) {
        self.permissions = permissions
        super.init(name: name, age: age)
    }
    
    override func greet() -> String {
        return "Hello, I'm admin \(name)"
    }
}
